go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/thanos-io/objstore
 
-go 1.20
+go 1.21
```

Comment:
Note, this one comes from latest `minio-go/v7`, which is not ideal, but shouldn't be a big deal

```
go: github.com/minio/minio-go/v7@v7.0.69 requires go@1.21, but 1.20 is requested
```